src/org/traccar/database/DataManager.java

```diff
@@ -115,6 +124,73 @@ private void initDatabase() throws Exception {
         }
     }
 
+    public static String constructObjectQuery(String action, Class<?> clazz, boolean additional) {
+        switch (action) {
+            case ACTION_INSERT:
+            case ACTION_UPDATE:
+                StringBuilder result = new StringBuilder();
+                StringBuilder fields = new StringBuilder();
+                StringBuilder values = new StringBuilder();
+
+                Set<Method> methods = new HashSet<>(Arrays.asList(clazz.getMethods()));
+                methods.removeAll(Arrays.asList(Object.class.getMethods()));
+                methods.removeAll(Arrays.asList(BaseModel.class.getMethods()));
+                for (Method method : methods) {
+                    if (method.getName().startsWith("get") && method.getParameterTypes().length == 0
+                            && (additional ? method.isAnnotationPresent(QueryAdditional.class)
+                                : !method.isAnnotationPresent(QueryIgnore.class)
+                                && !method.isAnnotationPresent(QueryAdditional.class))) {
```

Comment:
This if statement is really hard to read. I think you should split it. Also, why do we only check for ignore annotation only if additional is false?